File: 18_remove_substring.py
"""
Task 18: Remove a given substring from a string.
Reads two lines: main string and substring to remove. Prints result.
"""
import sys

if __name__ == '__main__':
    s = sys.stdin.readline().rstrip('\n')
    sub = sys.stdin.readline().rstrip('\n')
    print(s.replace(sub, ''))

# Time Complexity: O(n)
# Space Complexity: O(n)

File: 18_remove_substring.cpp
#include <bits/stdc++.h>
using namespace std;

/*
Task 18: Remove a given substring from a string.
Reads two lines: main string and substring to remove. Prints result.
*/

int main()
{
    string s, sub;
    if (!getline(cin, s))
        return 0;
    if (!getline(cin, sub))
        return 0;
    size_t pos = 0;
    while ((pos = s.find(sub, pos)) != string::npos)
    {
        s.erase(pos, sub.length());
    }
    cout << s << '\n';
    return 0;
}

// Time Complexity: O(n*m) worst-case depending on find
// Space Complexity: O(1)
